### ISSUE TITLE
How to use --exclude?

### ISSUE THREAD

**Issue by gajus** [webpro/knip#366]
Based on the description, I would expect this to not include `type` error:

```bash
$ knip --production --strict --ignore-internal --exclude type
Analyzing workspace ....
Unused exported types (1)
OrganizationExternalLink  type  src/contra-api/types.ts:2091:13
```

```
  --include                Report only provided issue type(s), can be comma-separated or repeated (1)
  --exclude                Exclude provided issue type(s) from report, can be comma-separated or repeated (1)
```

For context, we are running knip twice as follows:

```
knip && knip --production --strict --ignore-internal
```

and the issue is that the second run detects unused types that are contained in production files, but are only used in tests, therefore causing unused export type errors. It appears that the correct way to fix this is to just ignore unused export types in production build

**Comment by gajus** [comment]
That's probably why.

so... few suggestions for improvement here:

* should throw an error if unknown exclude member provided
* should align error messages with the type names
  * i.e. would expect that `OrganizationExternalLink  type  src/contra-api/types.ts:2091:13` says "types"

**Comment by webpro** [comment]
> should throw an error if unknown exclude member provided

Thanks for the suggestion, I like this idea.

> should align error messages with the type names
> i.e. would expect that OrganizationExternalLink  type  src/contra-api/types.ts:2091:13 says "types"

Although I understand your reasoning, they're not the same thing. The `types` issue group consists of `type`, `interface` and `enum`. The violation is about a single instance. Totally agree should definitely be documented more clearly!

**Comment by gajus** [comment]
Makes sense. Thank you for considering and for the explanation

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v3.1.0. See [Release 3.1.0](https://github.com/webpro/knip/releases/tag/3.1.0) for release notes._
